data-showcase/grails-app/services/nl/thehyve/datashowcase/ItemService.groovy

```diff
@@ -82,23 +94,29 @@ class ItemService {
                 ${dataShowcaseEnvironment.internalInstance ?
                         '' : 'where i.publicItem = true'
                 }
-                order by p.name asc
+                order by $property ${order == 'desc' ?
+                        'desc' : 'asc' 
+                }                
             """
         ).setResultTransformer(Transformers.ALIAS_TO_ENTITY_MAP)
+                .setFirstResult(firstResult)
+                .setMaxResults(maxResults)
                 .list() as List<Map>
         stopWatch.stop()
         stopWatch.start('Map to representations')
         def result = items.collect { Map itemData ->
             map(itemData)
         }
         stopWatch.stop()
-        log.info "Items fetched.\n${stopWatch.prettyPrint()}"
+        log.info "${result.size()} items fetched.\n${stopWatch.prettyPrint()}"
         result
     }
 
     @Transactional(readOnly = true)
-    List<ItemRepresentation> getItems(Set concepts, Set projects, SearchQueryRepresentation searchQuery) {
+    List<ItemRepresentation> getItems(int firstResult, int maxResults, String order, String propertyName,
```

Comment:
I guess this can be refactored to one `pageable` object, but that can be done later.